src/lib/hooks/use-device.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from '@tanstack/react-query'
+import type { Device, SeamError } from 'seamapi'
+
+import { useSeamClient } from 'lib/seam/use-seam-client.js'
+import type { UseSeamQueryResult } from 'lib/types/use-seam-query-result.js'
+
+export function useAccessCodes(
+  params:
+    | {
+        device_id: string
+      }
+    | string
+): UseSeamQueryResult<'device', Device<any>> {
```

Comment:
If we don't use the other req/res types:
```suggestion
): UseSeamQueryResult<'device', Device<CommonDeviceProperties>> {
```